src/software/amazon/ion/impl/lite/IonStructLite.java

```diff
@@ -483,6 +477,7 @@ protected void handle(IonValue newValue)
      */
     private void _add(String fieldName, IonValueLite child)
     {
+        if(!hasNullFieldName && fieldName == null) hasNullFieldName = true;
```

Comment:
Is there any benefit to checking `!hasNullFieldName` here? If it's already true, it's not harmful to set it to true.

Reply:
it will shortcircuit and prevent repeatedly setting hasNullFieldName

Reply:
Who cares if you repeatedly set it though?

Reply:
people paying for those clock cycles.

Reply:
A branch operation is probably similar in cost to (or even more expensive than) writing to an in-memory primitive with a known address. Both, though, are pretty trivial.

If you're really concerned with clock cycles, you can replace the whole if statement with:
```java
hasNullFieldName |= fieldName == null;
```
😄

Reply:
ohh nice